src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Flows/AcquireTokenHandlerBase.cs

```diff
@@ -220,12 +223,27 @@ public async Task<AuthenticationResult> RunAsync()
             catch (Exception ex)
             {
                 RequestContext.Logger.ErrorPii(ex);
-                if (_client != null && _client.Resiliency && extendedLifetimeResultEx != null)
+
+                if (_client == null)
+                {
+                    _client = new OAuthClient();
+                }
+
+                if (ex.InnerException is AdalServiceException serviceException)
                 {
-                    RequestContext.Logger.Info("Refreshing access token failed due to one of these reasons:- Internal Server Error, Gateway Timeout and Service Unavailable. " +
-                                       "Hence returning back stale access token");
+                    if (serviceException.StatusCode >= 500 && serviceException.StatusCode < 600 ||
+                    serviceException.StatusCode == 408)
```

Comment:
>408 [](start = 51, length = 3)

consider making this refer to a specific httpSTatus  code from the enum (timeout). I believe the enum already exists HttpStatusCode